src/Templates/TypeEditorTemplate/CustomTypeEditor/CustomControlLibrary.ClientServerCommunication/CustomControlLibrary.ClientServerCommunication.csproj

```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net7.0-windows;net472</TargetFrameworks>
+    <UseWindowsForms>true</UseWindowsForms>
+	<LangVersion>9.0</LangVersion>
```

Comment:
So, here is my thinking:
* It will be difficult to support 3.1, since there wouldn't Source Generators, and we will be needing them for the CollectionEditors.
* I am not sure, we should go down to support .NET 5, since a) it's out of support, and b) VB's application framework isn't really there at the time to fully support .NET. But for the latter, we need to have different templates anyway, and we can say, for VB it'll be .NET 6. If we then decided to support .NET 5 for Designer Support, we can't use C#10, because its minimal version is .NET 6. Up to now, I wanted support .NET 5 at least for C#, so we would be at C# 9.
* Then we're at .NET 6, and we could support C# 10.

Also, @MerrieMcGaw and @RussKie for FYI.

Reply:
I think I agree with that logic, as much as I want to say that we should support all versions forever - at the same time I want to use the most recent versions. This seems like a good compromise.

Reply:
So, after talking with a few additional folks, C#10, the completed VBAppFramework and .NET 6 are the minimal requirements to use these templates.